tls/extensions/s2n_npn.c

```diff
@@ -66,7 +66,31 @@ int s2n_server_npn_recv(struct s2n_connection *conn, struct s2n_stuffer *extensi
         return S2N_SUCCESS;
     }
 
-    POSIX_GUARD_RESULT(s2n_select_server_preference_protocol(conn, extension, supported_protocols));
+    /* Ignore errors as we can select our own protocol if parsing fails */
+    s2n_result_ignore(s2n_select_server_preference_protocol(conn, extension, supported_protocols));
+
```

Comment:
Idk about this. The RFC doesn't say that we should accept a non-empty malformed extension, just that we should accept an empty one. I think we need to still fail here.